File: Challenges/Coding World Cup/2023/Day 16 - AUS vs PAK - Supreme Bat.java
import java.util.*;

public class Solution {
    static boolean supreme(int w, int x, int y) {
        return w >= x && w <= y;
    }

    static int xAndY(int x, int y, int[] a) {
        List<Integer> s = new ArrayList<>();
        for (int i = 0; i < a.length; i++) {
            if (supreme(a[i], x, y)) {
                s.add(a[i]);
            }
        }

        Collections.sort(s);

        if (s.size() == 0) {
            return -1;
        }
        return s.get(0);
    }
}

File: Challenges/Coding World Cup/2023/Day 16 - AUS vs PAK - Supreme Bat.cpp
bool supreme(int w, int x, int y)
{
    return w >= x && w <= y;
}

int xAndY(int x, int y, vector<int> &a)
{
    // Write your code here.
    vector<int> s;
    for (int i = 0; i < a.size(); i++)
    {
        if (supreme(a[i], x, y))
            s.push_back(a[i]);
    }

    sort(s.begin(), s.end());

    if (s.size() == 0)
        return -1;
    return s[0];
}

